pages/avatar/[address].tsx

```diff
@@ -375,6 +394,178 @@ const Badge = (props: { item: Inventory.AcceptedInventoryItem }) => {
+    notificationStore.pushNotification("Minting Avatar.", {
+      type: "Info",
+      autoRemove: true,
+    });
+    notificationStore.removeNotification;
+    const response = await Avatar.claim(avatarSdk, props.address);
+    if (!response?.avatar) {
+      throw new Error((response as any).message);
+    }
+    notificationStore.pushNotification("Avatar successfully minted!", {
+      type: "Success",
+    });
+    props.onClaimSuccess();
+    modalStore.closeModal();
+  };
+
+  const onClickBurn = async () => {
+    setIsClaiming(true);
+    try {
+      if (props.hasCrossed) {
+        try {
+          await doClaim();
+        } catch (error) {
+          notificationStore.pushNotification(error.message, {
+            type: "Error",
+          });
+        }
+        setIsClaiming(false);
+      } else {
+        const signer = store.wallets.getActiveSigner() as ExtSigner;
+        await signAndSend(
+          tx,
+          signer,
+          extrinsicCallback({
+            notificationStore,
+            broadcastCallback: () => {
+              notificationStore.pushNotification("Burning ZTG.", {
+                type: "Info",
+                autoRemove: true,
+              });
+            },
+            successCallback: async () => {
+              try {
+                await delay(2000);
+                await doClaim();
+                setIsClaiming(false);
+                props.onClose?.();
+              } catch (error) {
+                notificationStore.pushNotification(error.message, {
+                  type: "Error",
+                });
+              }
+            },
+            retractedCallback: async () => {
+              setIsClaiming(false);
+            },
+            failCallback: ({ index, error }) => {
+              setIsClaiming(false);
+              notificationStore.pushNotification(
+                store.getTransactionError(index, error),
+                { type: "Error" },
+              );
+            },
+          }),
+        );
+      }
+    } catch (error) {
+      setIsClaiming(false);
+    }
+  };
+
+  useEffect(() => {
+    tx.paymentInfo(props.address).then((fee) =>
+      setFee(fee.partialFee.toJSON()),
+    );
+  }, [tx, props.address]);
+
+  useEffect(() => {
+    return () => {
+      props.onClose?.();
+    };
+  }, [props.onClose]);
+
+  return (
+    <div>
+      <div className="flex flex-1">
+        <div className="pr-6">
+          <p className="mb-4">
+            To claim your right to mint an avatar you have to pay the ferryman
+            due respect, burning {props.burnAmount / ZTG} ZTG.
+          </p>
+          {!props.hasCrossed ? (
+            <div className="flex items-center">
+              <div className="text-red-800 mr-4">
+                <IoIosWarning size={22} />
+              </div>
+              <div className="text-red-800 text-xs flex-1">
+                The amount will be burned(slashed) and not paid to any address.
```

Comment:
This quite a lot text that I think we can avoid. We could show something like 'Amount to burn {burnAmount}' at the top then validate their balance so they don't have to check. Then we can display a message if there account is at risk of being reaped based on their remaining balance